language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -528,92 +570,74 @@ writePackageIdTs dir pkgId =
+        , "module" .= ("commonjs" :: T.Text)
+        , "declaration" .= True
+        , "sourceMap" .= True ]
+      , "include" .= (["src/**/*.ts"] :: [T.Text])
+      ]
 
 writeEsLintConfig :: FilePath -> IO ()
-writeEsLintConfig dir = writeFileUTF8 (dir </> ".eslintrc.json") $ unlines
-  [ "{"
-  , "  \"parser\": \"@typescript-eslint/parser\","
-  , "  \"parserOptions\": {"
-  , "    \"project\": \"./tsconfig.json\""
-  , "  },"
-  , "  \"plugins\": ["
-  , "    \"@typescript-eslint\""
-  , "  ],"
-  , "  \"extends\": ["
-  , "    \"eslint:recommended\","
-  , "    \"plugin:@typescript-eslint/eslint-recommended\","
-  , "    \"plugin:@typescript-eslint/recommended\","
-  , "    \"plugin:@typescript-eslint/recommended-requiring-type-checking\""
-  , "  ],"
-  , "  \"rules\": {"
-  , "    \"@typescript-eslint/explicit-function-return-type\": \"off\","
-  , "    \"@typescript-eslint/no-inferrable-types\": \"off\""
-  , "  }"
-  , "}"
-  ]
+writeEsLintConfig dir =
+  BSL.writeFile (dir </> ".eslintrc.json") $ encodePretty esLintConfig
+  where
+    esLintConfig :: Value
+    esLintConfig = object [
+        "parser" .= ("@typescript-eslint/parser" :: T.Text)
+      , "parserOptions" .= object [("project", "./tsconfig.json")]
+      , "plugins" .= (["@typescript-eslint"] :: [T.Text])
+      , "extends" .= ([
+             "eslint:recommended"
+           , "plugin:@typescript-eslint/eslint-recommended"
+           , "plugin:@typescript-eslint/recommended"
+           , "plugin:@typescript-eslint/recommended-requiring-type-checking"
+           ] :: [T.Text])
+      , "rules" .= object [
+            ("@typescript-eslint/explicit-function-return-type", "off")
+          , ("@typescript-eslint/no-inferrable-types", "off") ]
+      ]
 
 writePackageJson :: FilePath -> SdkVersion -> Scope -> [Dependency] -> IO ()
 writePackageJson packageDir sdkVersion (Scope scope) depends =
-  writeFileUTF8 (packageDir </> "package.json") $ unlines
-  (["{"
-   , "  \"private\": true,"
-   , "  \"name\": \"" <> name <> "\","
-   , "  \"version\": \"" <> version <> "\","
-   , "  \"description\": \"Produced by daml2ts\","
-   , "  \"dependencies\": {"
-   , "    \"@daml/types\": \"" <> version <> "\","
-   ] ++ dependencies ++
-   [ "    \"@mojotech/json-type-validation\": \"^3.1.0\""
-    , "  },"
-    , "  \"scripts\": {"
-    , "    \"build\": \"tsc --build\","
-    , "    \"lint\": \"eslint --ext .ts src/ --max-warnings 0\""
-    , "  },"
-    , "  \"devDependencies\": {"
-    , "    \"@typescript-eslint/eslint-plugin\": \"^2.11.0\","
-    , "    \"@typescript-eslint/parser\": \"^2.11.0\","
-    , "    \"eslint\": \"^6.7.2\","
-    , "    \"typescript\": \"~3.7.3\""
-    , "  }"
-    , "}"
-    ])
+  BSL.writeFile (packageDir </> "package.json") $
+    encodePretty (packageJson (NpmPackageName name) (NpmPackageVersion version) dependencies)
   where
-    version = versionToString sdkVersion
+    version = versionToText sdkVersion :: T.Text
```

Comment:
There's no need for a type annotation since the function is completely monomorphic.
```suggestion
    version = versionToText sdkVersion
```

Reply:
must have been left over from debugging